conda_smithy/lint_recipe.py

```diff
@@ -290,10 +290,8 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
 
     # 12a: License family must be valid (conda-build checks for that)
     license_family = about_section.get("license_family", license).lower()
-    license_file = about_section.get("license_file", "")
-    if license_file == "" and any(
-        f for f in NEEDED_FAMILIES if f in license_family
-    ):
+    license_file = about_section.get("license_file", None)
+    if not license_file:
```

Comment:
Should this use `NEEDED_FAMILIES` or should it require everything to have a licence file?

If `NEEDED_FAMILIES` is desirable:

```suggestion
    if not license_file and any(
        f for f in NEEDED_FAMILIES if f in license_family
     ):
```